core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -659,6 +618,132 @@ class BrokerTopicStats(configOpt: java.util.Optional[KafkaConfig] = java.util.Op
     }
   }
 
+  // Update the broker-level all topic metric values so that we have a sample right for all topics metric after update of partition.
+  def recordRemoteCopyLagBytes(topic: String, partition: Int, value: Long): Unit = {
+    val topicMetric = topicStats(topic)
+    topicMetric.remoteCopyLagBytesAggrMetric().setValue(String.valueOf(partition), value)
+    allTopicsStats.remoteCopyLagBytesAggrMetric().setValue(topic, topicMetric.remoteCopyLagBytes)
+  }
+
+  // Update the broker-level all topic metric values so that we have a sample right for all topics metric after removal of partition.
+  def removeRemoteCopyLagBytes(topic: String, partition: Int): Unit = {
+    val topicMetric = topicStats(topic)
+    topicMetric.remoteCopyLagBytesAggrMetric().removeKey(String.valueOf(partition))
+    allTopicsStats.remoteCopyLagBytesAggrMetric().setValue(topic, topicMetric.remoteCopyLagBytes)
+  }
```

Comment:
After refactor, the per-topic and per-broker metric will be updated within one method to avoid forgetting about it.